### ISSUE TITLE
Support for Handlebars 3.0.0?

### ISSUE THREAD

**Issue by burkhartt** [lazd/gulp-handlebars#47]
Does gulp-handlebars support handlebars 3.0.0? I am getting the following error after upgrading from v2 to v3:

```
Uncaught TypeError: Cannot assign to read only property 'partial' of categoryForm

handlebars-v3.0.0.js:652
invokePartialhandlebars-v3.0.0.js:496
invokePartialWrappertemplates.js:51
Adventures.templates.categoryUpdate.Handlebars.template.mainhandlebars-v3.0.0.js:593
retapp.min.js:56
Component.getTemplateapp.min.js:51
(anonymous function)app.min.js:961
CategoryUpdate.resolveDeferredModel.$.ajax.completejquery-1.11.2.min.js:2
m.Callbacks.jjquery-1.11.2.min.js:2
m.Callbacks.k.fireWithjquery-1.11.2.min.js:4
xjquery-1.11.2.min.js:4 m.ajaxTransport.send.b
```

This is running in Chrome. A strange thing is that Safari is actually running this fine...

**Comment by ubaltaci** [comment]
Did you compile like [there](https://github.com/lazd/gulp-handlebars#compiling-using-a-specific-handlebars-version) with your own version of handlebars ?

**Comment by DPflasterer** [comment]
+1

**Comment by lazd** [comment]
Yes, `gulp-handlebars` works nicely with Handlebars@3.0.0. I just modified the example for Handlebars@1.x and it compiled and ran just fine.

Looking at your stack trace, you're trying to assign the `partial` property of `categoryForm`, which is somehow read-only. You can make properties read-only in JavaScript using [property descriptors and Object.defineProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty), is it possible you've done that somewhere?

Given that it's working in one browser and not the other, I I don't think this is a problem with `gulp-handlebars` in any case, it's a runtime error. 

I'm going to close this issue for now. If you still think `gulp-handlebars` is causing your issue, please re-open and include the build output and a simplified example that reproduces the problem reliably.

**Comment by lazd** [comment]
I pushed a [`handlebars3` example](https://github.com/lazd/gulp-handlebars/tree/master/examples/handlebars3) you can play with. If you re-open this issue, please modify the `handlebars3` example such that your issue is reproduced, then push it to your fork and link to it from here so we can debug it. Thanks!

**Comment by DPflasterer** [comment]
That's because you're not using a partial in your handlebars1/3 examples. Try again, only this time update your partials example to use 3.0.0.

**Comment by lazd** [comment]
@DPflasterer I was able to reproduce a similar error by doing what you said, I'm looking into it now.

**Comment by lazd** [comment]
@DPflasterer @burkhartt I've pushed another full example, [handlebars3-partials](https://github.com/lazd/gulp-handlebars/tree/master/examples/handlebars3-partials) showing how to use `gulp-handlebars` with `handlebars@3.0.0`.

The bug in question was happening because the `options.handlebars` property was not passed to `gulp-handlebars`, so it was compiling with `handlebars@2.x` instead of `handlebars@3.x`. If you remove [this line in the `handlebars3-partials` example](https://github.com/lazd/gulp-handlebars/blob/ae594dfad232a7c0bc5bd7739b0b01d07055dc3a/examples/handlebars3-partials/gulpfile.js#L15), you will see the bug as reported in this issue.

### To use `handlebars@3.0.0`, you need to pass the Handlebars module you want to use as [`options.handlebars` as noted in the documentation](https://github.com/lazd/gulp-handlebars#optionshandlebars)

**Comment by lazd** [comment]
`gulp-handlebars@4.0.0` has been released and bundles `handlebars@3.0.0` by default. I hope this clears up confusion for new users of this tool. Know that you can always use whatever version of Handlebars you like by passing it as [`options.handlebars`](https://github.com/lazd/gulp-handlebars#optionshandlebars).

If you upgrade to `gulp-handlebars@4.0.0` and still need to use `handlebars@2.0.0`, please follow the [`handlebars2`](https://github.com/lazd/gulp-handlebars/examples/handlebars2) and [`handlebars2-partials`](https://github.com/lazd/gulp-handlebars/examples/handlebars2-partials) examples.
